app/src/panels/RequestSalary/AllocationFields.js

```diff
@@ -0,0 +1,222 @@
+import React, { useCallback } from 'react'
+import styled from 'styled-components'
+import BN from 'bn.js'
+import {
+  GU,
+  Slider,
+  TextInput,
+  textStyle,
+  tokenIconUrl,
+  useTheme,
+} from '@aragon/ui'
+
+import { toDecimals } from '../../utils/math-utils'
+import { formatTokenAmount } from '../../utils/formatting'
+import { splitAllocation } from '../../utils/calculations'
+
+const ONE_PCT = new BN(toDecimals('1', 16))
+
+function AllocationFields({
+  amount,
+  baseAsset,
+  baseAssetAllocation,
+  equityAsset,
+  equityMultiplier,
+  onAllocationChange,
+  pctBase,
+}) {
+  const theme = useTheme()
+
+  // Allocation changed from base asset input
+  const handleBaseAllocationChange = useCallback(
+    event => {
+      const baseAllocation = Math.min(Math.max(0, event.target.value), 100)
+      const baseAllocationBN = ONE_PCT.mul(new BN(baseAllocation))
+
+      onAllocationChange(baseAllocationBN)
+    },
+    [onAllocationChange]
+  )
+
+  // Allocation changed from equity asset input
+  const handleEquityAllocationChange = useCallback(
+    event => {
+      const equityAllocation = Math.min(Math.max(0, event.target.value), 100)
+      const baseAllocation = 100 - equityAllocation
+      const baseAllocationBN = ONE_PCT.mul(new BN(baseAllocation))
+
+      onAllocationChange(baseAllocationBN)
+    },
+    [onAllocationChange]
+  )
+
+  // Allocation changed from slider
+  const handleSliderChange = useCallback(
+    value => {
+      const baseAllocation = (value * 100).toFixed()
+      const baseAllocationBN = ONE_PCT.mul(new BN(baseAllocation))
+
+      onAllocationChange(baseAllocationBN)
+    },
+    [onAllocationChange]
+  )
+
+  const BASE_ASSET_COLOR = theme.blue
+  const EQUITY_ASSET_COLOR = theme.red
+
+  // Convert allocation to %
+  const [convertedBaseAllocation, convertedEquityAllocation] = splitAllocation(
+    baseAssetAllocation,
+    pctBase
+  )
+
+  // Calculate base asset and equity asset total amounts given the allocation used
+  // Note that the equity total amount is represented in base asset decimals
+  const baseAssetTotalAmount = amount.mul(baseAssetAllocation).div(pctBase)
+  const equityAssetTotalAmount = amount
+    .sub(baseAssetTotalAmount)
+    .mul(equityMultiplier.div(pctBase))
+
+  return (
+    <div>
+      <div
+        css={`
+          display: flex;
+          align-items: center;
+          justify-content: space-between;
+          margin-bottom: ${1 * GU}px;
+        `}
+      >
+        <AllocationInput
+          text="Base asset percentage"
+          value={convertedBaseAllocation}
+          onChange={handleBaseAllocationChange}
+        />
+        <AllocationInput
+          text="Equity asset percentage"
+          value={convertedEquityAllocation}
+          onChange={handleEquityAllocationChange}
+          css="text-align:right"
+        />
+      </div>
+      <CostumSlider
```

Comment:
i think this component name was intended to be `CustomSlider`? haha

Reply:
haha yeah 😅